vizro-ai/src/vizro_ai/_vizro_ai.py

```diff
@@ -91,17 +91,18 @@ def _run_plot_tasks(
             "code_string": code_string,
         }
 
-    def _get_chart_code(self, df: pd.DataFrame, user_input: str) -> str:
+    def _get_chart_code(self, df: pd.DataFrame, user_input: str, explain: bool = False) -> str:
         """Get Chart code of vizro via english descriptions, English to chart translation.
 
         Can be used in integration with other application if only code snippet return is required.
 
         Args:
             df: The dataframe to be analyzed
             user_input: User questions or descriptions of the desired visual
+            explain: Flag to include explanation in response
         """
         # TODO refine and update error handling
-        return self._run_plot_tasks(df, user_input, explain=False).get("code_string")
+        return self._run_plot_tasks(df, user_input, explain=explain)
```

Comment:
this _get_chart_code is specifically for application integration, just for get code. 
- 'if only code snippet return is required.' is the only requirement. 
- if user wants to have explanation, this is equal to use  _run_plot_tasks call
- better to just keep one call if that is the case

Reply:
Hm good point. In this case a user actually ran it with `explain` but I think this may be mainly because it was wrong in the example. I will revert then to code only return, but maybe refer to the other function in the docs.